src/views/templates/details/tabs/details/components/WorkloadProfile.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import produce from 'immer';
+import useWorkloadProfile, {
+  WorkloadProfileKeys,
+} from 'src/views/templates/list/hooks/useWorkloadProfile';
+import { ensurePath } from 'src/views/templates/utils';
+
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template';
+import { VM_WORKLOAD_ANNOTATION } from '@kubevirt-utils/resources/vm';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+
+import { TemplateDetailsGridProps } from '../TemplateDetailsPage';
+
+import WorkloadProfileModal from './WorkloadProfileModal';
+
+const WorkloadProfile: React.FC<TemplateDetailsGridProps> = ({ template, editable }) => {
+  const { createModal } = useModal();
+  const { t } = useKubevirtTranslation();
+  const workload = useWorkloadProfile(template, t);
+
+  const updateWorkload = (updatedWorkload: WorkloadProfileKeys) => {
+    const updatedTemplate = produce<V1Template>(template, (templateDraft: V1Template) => {
+      const vm = getTemplateVirtualMachineObject(templateDraft);
+      ensurePath(vm, 'spec.template.metadata.annotations');
+
+      vm.spec.template.metadata.annotations[VM_WORKLOAD_ANNOTATION] = updatedWorkload;
+
+      return templateDraft;
+    });
+
+    return k8sUpdate({
```

Comment:
maybe `k8sPatch` makes more sense here ?

Reply:
Maybe, but works fine also with the `k8sUpdate`.

Reply:
When using _k8sPatch_, I need to know the exact path, where I want to change the data -  that can be a bit difficult to solve when I change something in the Template's VM object. Recently you were explaining to me that the VM object can be anywhere, not only/always at the first place of the _template.objects_ array. Using _k8sPatch_ I'd have to solve this somehow - I consider this as unnecessary, and so `k8sUpdate` is much easier and straightforward to use here.

Reply:
yes, you will need to first find the index of the vm object and only then use patch

Reply:
There are more issues to solve than only to find the index of the VM object...